examples/js/loaders/STLLoader.js

```diff
@@ -254,53 +280,78 @@ THREE.STLLoader.prototype = {
 
 			var result;
 
-			while ( ( result = patternFace.exec( data ) ) !== null ) {
-
-				var vertexCountPerFace = 0;
-				var normalCountPerFace = 0;
+			var groupVertexes = [];
+			var groupCount = 0;
+			var startVertex = 0;
+			var endVertex = 0;
 
-				var text = result[ 0 ];
+			while ((result = patternSolid.exec(data)) !== null) {
 
-				while ( ( result = patternNormal.exec( text ) ) !== null ) {
+				startVertex = endVertex;
 
-					normal.x = parseFloat( result[ 1 ] );
-					normal.y = parseFloat( result[ 2 ] );
-					normal.z = parseFloat( result[ 3 ] );
-					normalCountPerFace ++;
+				solid = result[0];
 
-				}
-
-				while ( ( result = patternVertex.exec( text ) ) !== null ) {
-
-					vertices.push( parseFloat( result[ 1 ] ), parseFloat( result[ 2 ] ), parseFloat( result[ 3 ] ) );
-					normals.push( normal.x, normal.y, normal.z );
-					vertexCountPerFace ++;
+				while ( ( result = patternFace.exec( solid ) ) !== null ) {
 
+					var vertexCountPerFace = 0;
+					var normalCountPerFace = 0;
+		
+					var text = result[ 0 ];
+		
+					while ( ( result = patternNormal.exec( text ) ) !== null ) {
+		
+						normal.x = parseFloat( result[ 1 ] );
+						normal.y = parseFloat( result[ 2 ] );
+						normal.z = parseFloat( result[ 3 ] );
+						normalCountPerFace ++;
+		
+					}
+		
+					while ( ( result = patternVertex.exec( text ) ) !== null ) {
+		
+						vertices.push( parseFloat( result[ 1 ] ), parseFloat( result[ 2 ] ), parseFloat( result[ 3 ] ) );
+						normals.push( normal.x, normal.y, normal.z );
+						vertexCountPerFace ++;
+						endVertex++;
+		
+					}
+		
+					// every face have to own ONE valid normal
+		
+					if ( normalCountPerFace !== 1 ) {
+		
+						console.error( 'THREE.STLLoader: Something isn\'t right with the normal of face number ' + faceCounter );
+		
+					}
+		
+					// each face have to own THREE valid vertices
+		
+					if ( vertexCountPerFace !== 3 ) {
+		
+						console.error( 'THREE.STLLoader: Something isn\'t right with the vertices of face number ' + faceCounter );
+		
+					}
+		
+					faceCounter ++;
+		
 				}
 
-				// every face have to own ONE valid normal
+				groupVertexes.push({ startVertex: startVertex, endVertex: endVertex });
+				groupCount++;
 
-				if ( normalCountPerFace !== 1 ) {
-
-					console.error( 'THREE.STLLoader: Something isn\'t right with the normal of face number ' + faceCounter );
-
-				}
-
-				// each face have to own THREE valid vertices
+			}
+				
+			geometry.addAttribute( 'position', new THREE.Float32BufferAttribute( vertices, 3 ) );
+			geometry.addAttribute( 'normal', new THREE.Float32BufferAttribute( normals, 3 ) );
 
-				if ( vertexCountPerFace !== 3 ) {
+			if (groupCount > 0) {
```

Comment:
Some code style issues. Please ensure to add spaces after/before brackets like so:
```js
if ( groupCount > 0 ) {
```

Reply:
Corrected in commit 3ea613a.

Reply:
Please try using https://zz85.github.io/mrdoobapproves/ to find the remaining formatting issues.

Reply:
Is there a style sheet to be used in editors for this?

Reply:
I have applied the changes made by the tool you linked (see commit 9ece2588c2f515de3a373991f709875c24c0dd24), which now gives a green light. 
However, it changed also some parts I have not touched (e.g. spacing after function name at line 145, space before ++ (ok, this hurts my C eyes :-)). Let me know if that is OK.

Reply:
> Is there a style sheet to be used in editors for this?

When you an editor like VSCode or Atom, you should have automatically linter support by importing the `three.js` repository as a project and installing the `dev` dependencies via `npm i`.

Reply:
Thank you @Mugen87! I indeed used VSCode. I'll look into get it to work for the future.